### ISSUE TITLE
loss does not decrease in Google Colab and local computer. (TF.Keras 1.13 vs from TF.Keras 1.14 to TF.Keras 2.6)

### ISSUE THREAD

**Issue by HeewonChung92** [keras-team/keras#16645]
<details><summary>Click to expand!</summary> 
 
 ### Issue Type

Performance

### Source

source

### Tensorflow Version

1.13 vs 2.6

### Custom Code

Yes

### OS Platform and Distribution

Windows 10

### Python version

3.7.4

### CUDA/cuDNN version

11.2.0 / 8.1.0

### Current Behaviour?
I asked this issue to tensorflow, but tensorflow told me to ask keras repo.
https://github.com/tensorflow/tensorflow/issues/56080

This is my full source code!
https://colab.research.google.com/drive/1tSE7t3GPa8r4x7F4sPgj8CqiYqtOrej5?usp=sharing
https://drive.google.com/drive/folders/17Nnl5FcrUWnRgzz4DM_fNAI3KhwqcK3r?usp=sharing

```shell
I studied with my data and tf-gpu 1.6 and keras in 2019. 
At this time, training was done normally.

Although it is the same code as in the past, 
training loss and validation loss are decreased 
when training with TF 1.13  (working TF 1.6 ~ TF 1.13).

However, from TF 1.14 to TF 2.6, 
both train loss and validation loss have increased.

When the model weight made with TF 1.6 was called from TF 2.6 (tried to TF 1.14, TF 1.15, TF 2.0 and TF 2.6), 
it was called without any problems and the results were the same as those from the past.

So I can't understand why that happen.
I wonder what parts of TF 1.13 and TF 1.14 and above are different and affect training!!!!!!

Also, I tried updating the source code through tf_upgrade_v2, 
but it was not solved at all.
https://www.tensorflow.org/guide/upgrade?hl=en
tf_upgrade_v2 --infile my_code.py --outfile my_code_upgraded.py
```

### Relevant log output

**decreased the loss when using TF 1.13**
![image](https://user-images.githubusercontent.com/39148033/168065098-12939b28-dbe7-4cef-9cb3-ecb0ea18843c.png)

**increased the loss when using TF 1.14 and TF 2.6**
![image](https://user-images.githubusercontent.com/39148033/168066373-cb5bb8b4-1f18-41ba-a412-446c5d0dd195.png)

_No response_</details></details>

**Comment by HeewonChung92** [comment]
@gowthamkpr Hello. How are you? Is there any problem with my code?

**Comment by gowthamkpr** [comment]
@HeewonChung92 Can you please create a colab notebook and attach it here..!! Thanks!

**Comment by HeewonChung92** [comment]
@gowthamkpr Here is my code!!! https://colab.research.google.com/drive/1tSE7t3GPa8r4x7F4sPgj8CqiYqtOrej5?usp=sharing
https://drive.google.com/drive/folders/17Nnl5FcrUWnRgzz4DM_fNAI3KhwqcK3r?usp=sharing

**Comment by gowthamkpr** [comment]
@HeewonChung92 Sorry for the inconvenience. Can you please install different versions of tensorflow and run the code with different versions and attach the gists here. Also can you please use the latest versions of tensorflow. Thanks!

**Comment by HeewonChung92** [comment]
@gowthamkpr  I attach the colab code here with two versions.   

Code - Tensorflow version: 1.13.2   
decreased the loss when using TF 1.13.2  
https://colab.research.google.com/drive/1f-eUOLIDTESbWjw0WDmZjl-D65KwGT8H?usp=sharing

Code - Tensorflow version: 2.8 (latest version)   
increased the loss when using TF 1.14 and TF 2.6   
https://colab.research.google.com/drive/1uh9VMLrMcFJUAlCwDkbRBdiVclcLoKiL?usp=sharing

**Comment by gowthamkpr** [comment]
@HeewonChung92 Thank you! Will look into this

**Comment by gowthamkpr** [comment]
@HeewonChung92 Can you debug the model layer by layer and check where the issue is coming from. Thanks

**Comment by HeewonChung92** [comment]
@gowthamkpr
 I think it will take some time because I have a lot of work to do. I will check and let you know which layer is the problem.

**Comment by gowthamkpr** [comment]
@HeewonChung92 Please let me know if you were able to debug it layer by layer as mentioned above. Thanks!
